### ISSUE TITLE
pre-push hook not showing output

### ISSUE THREAD

**Issue by bencresty** [git-for-windows/git#3419]
- [ x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

![image](https://user-images.githubusercontent.com/69322733/132705559-346c46b7-c744-4bd5-9a5b-2557849cfab9.png)

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

![image](https://user-images.githubusercontent.com/69322733/132705744-2766f358-8520-4143-b2e2-5b0a205b9a36.png)

 - What options did you set as part of the installation? Or did you choose the
   defaults?

pretty much the basic config except for basic stuff like username/email/editor/sslcainfo

```
type "C:\Program Files\Git\etc\install-options.txt"
```

![image](https://user-images.githubusercontent.com/69322733/132706355-100ef3e5-715a-4e58-a4e3-0a756d8af96c.png)

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

no

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Mostly VS Code

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

Using `git push` 

 - What did you expect to occur after running these commands?

The `pre-push` hook to be executed. But it's not running.

 - What actually happened instead?

The script isn't running.

- What does your `pre-push` script look like?

I tried different things like triggering a windows batch file or just echoing a message. These things work in the same repository from within other hooks, but not from the `pre-push` hook.

**Example A)** which is working in hooks like `pre-commit`, `prepare-commit` and `commit-msg`, but NOT from `pre-push`:

``` bash
#!/bin/sh

echo '------------- PRE-PUSH -------------------'
cmd.exe /c "batch-files\commit-msg.bat"
exit 0
```

**Example B)** which is working in hooks like `pre-commit`, `prepare-commit` and `commit-msg`, but NOT from `pre-push`:

``` bash
#!/bin/sh

cmd.exe /c "batch-files\pre-push.bat"
exit 0
```

where `batch-files\pre-push.bat` is like:

``` bat
echo '=============== PRE-PUSH BATCH ============='
```

**Comment by bencresty** [comment]
You're right. When I do exactly what you do in a command line and without an actual origin on a remote it works here too. 

So I just created yet another fresh repository on bitbucket and cloned it locally, added the `pre-push` hook as before and commited and pushed it via VS Code with the build in push.
This does a lot of things at once under the hood, but never shows the output of the `pre-push` script and although mentioned doesn't seem to be fired:

![push-hook-not-working-vs-code](https://user-images.githubusercontent.com/69322733/132728892-50614247-bf48-4392-933b-8433b74d902a.png)


Which is weird to me, because when commiting from VS Code's build in 'commit' function it DOES output the commit hooks in the same output panel:

![commit-vs-code-does-work](https://user-images.githubusercontent.com/69322733/132729118-2f8602a9-9bd8-4a0e-bd96-02088edf19a3.png)


So I tried to push in the same project from the command line by just doing a simple push (not alle the commands VS Code is firing when only clicking 'push'):

![it-works-from-command-line](https://user-images.githubusercontent.com/69322733/132729459-2744a57b-e645-47bf-9260-c86547d3cd6b.png)


I know a few things about git, but I'm definitely not an expert. Any clue on what could cause the VS Code `push` command, which does a `git push` underwater, to prevent the script being triggered?

**Comment by dscho** [comment]
Best guess: use Process Monitor to capture the exact `git push` command line. It might have disabled the hook specifically.

**Comment by bencresty** [comment]
@dscho sorry, I don't know Process Monitor. I found Process Monitor from Microsoft. Is that what you mean?
https://docs.microsoft.com/en-us/sysinternals/downloads/procmon

I just downloaded it and have it open. But not sure how I could debug this `git push` line in there other than in the `git` outputpanel of VS Code itself, which seems to show all lines it fires to git, like the `git push origin master:master` command line.
It's not clear to me what else would be to see about the command lines that aren't in the git output panel.

Could you please help me on what you mean and how to debug what you have in mind with Process Monitor?

**Comment by bencresty** [comment]
Alright, I just did a test from VS Code using the `pre-push` hook, but now replaced the echo command with a call to a batch-file which isn't only echoing, but also creates a new folder.

pre-push.bat:
``` text
@echo --------------- BATCH FILE EXECUTING ----------------

mkdir test-created-by-push
```

When running the push from VS Code now I can see the `pre-push` hook DOES fire because the folder is being created by the batch file when git pushing via VS Code.

So that narrows the problem to VS Code not displaying the output of the hook in the output panel, eventhough it DOES output the output of the commit-hooks. So for some weird reason there is a difference between which hook is running if VS Code outputs it or not.

I just found some threads about this of at least one person having this issue too before. But the issue is closed, so should tell us this should be solved. But it's clearly not solved, or I am missing some setting now behind some flag somewhere. Of which I wouldn't know why it would be turned off in the first place.

It seems to be no issue with Git for Windows though, so I'll continue my search on the VS Code side of things now and leave this open here for now, because I'm not 100% sure on this.
